benchmarks/triangulated_surface_distance/triangulated_surface_distance.cpp

```diff
@@ -0,0 +1,312 @@
+
+  constexpr int DIM =
+      ArborX::GeometryTraits::dimension_v<typename Points::value_type>;
+
+  auto vertices_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, vertices);
+  auto triangles_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, triangles);
+
+  std::ofstream out(filename);
+
+  out << "# vtk DataFile Version 2.0\n";
+  out << "Mesh example\n";
+  out << "ASCII\n";
+  out << "DATASET POLYDATA\n\n";
+  out << "POINTS " << num_vertices << " float\n";
+  for (int i = 0; i < num_vertices; ++i)
+  {
+    for (int d = 0; d < DIM; ++d)
+      out << " " << vertices(i)[d];
+    out << '\n';
+  }
+
+  int const num_cell_vertices = 3;
+  out << "\nPOLYGONS " << num_elements << " "
+      << (num_elements * (1 + num_cell_vertices)) << '\n';
+  for (int i = 0; i < num_elements; ++i)
+  {
+    out << num_cell_vertices;
+    for (int j = 0; j < num_cell_vertices; ++j)
+      out << " " << triangles(i * num_cell_vertices + j);
+    out << '\n';
+  }
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  std::cout << "ArborX version    : " << ArborX::version() << '\n';
+  std::cout << "ArborX hash       : " << ArborX::gitCommitHash() << '\n';
+  std::cout << "Kokkos version    : " << ArborX::Details::KokkosExt::version()
+            << '\n';
+
+  namespace bpo = boost::program_options;
+
+  bpo::options_description desc("Allowed options");
+  int n;
+  int num_refinements;
+  float radius;
+  std::string vtk_filename;
+  // clang-format off
+  desc.add_options()
+      ( "help", "help message" )
+      ( "n", bpo::value<int>(&n)->default_value(1000), "number of points" )
+      ( "radius", bpo::value<float>(&radius)->default_value(1.f), "sphere radius" )
+      ( "refinements", bpo::value<int>(&num_refinements)->default_value(5), "number of icosahedron refinements" )
+      ( "vtk-filename", bpo::value<std::string>(&vtk_filename), "filename to dump mesh to in VTK format" )
+      ;
+  // clang-format on
+  bpo::variables_map vm;
+  bpo::store(bpo::command_line_parser(argc, argv).options(desc).run(), vm);
+  bpo::notify(vm);
+
+  if (vm.count("help") > 0)
+  {
+    std::cout << desc << '\n';
+    return 1;
+  }
+
+  auto [vertices_v, triangles_v] = buildTriangles(radius, num_refinements);
+
+  auto vertices = vec2view<MemorySpace>(vertices_v);
+  auto triangles = vec2view<MemorySpace>(triangles_v);
+
+  if (!vtk_filename.empty())
+    writeVtk(vtk_filename, vertices, triangles);
+
+  ExecutionSpace space;
+
+  Kokkos::Profiling::pushRegion("Benchmark::build_points");
+  Kokkos::View<Point *, MemorySpace> random_points(
+      Kokkos::view_alloc(space, Kokkos::WithoutInitializing,
+                         "Benchmark::points"),
+      n);
+  generatePointCloud(PointCloudType::filled_box, std::cbrt(n), random_points);
```

Comment:
How do you control the volume in which points are being generated?

Reply:
I don't. I generate the points in `std::cbrt(n)` box. I thought about providing an argument for it, but I honestly don't see what it adds to the benchmark.

Reply:
I suppose not much as long as it stays centered on the origin